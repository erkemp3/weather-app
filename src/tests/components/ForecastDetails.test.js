import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 25,
    wind: {
      speed: 15,
      direction: "s",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Thursday 1st January 1970")).toHaveClass(
      "forecast-details__date"
    );
    expect(getByText("Max temp: 22°C")).toHaveClass(
      "forecast-details__temperature"
    );
    expect(getByText("Min temp: 12°C")).toHaveClass(
      "forecast-details__temperature"
    );
    expect(getByText("Humidity: 25%")).toHaveClass(
      "forecast-details__humidity"
    );
    expect(getByText("Wind speed: 15mph")).toHaveClass(
      "forecast-details__wind"
    );
    expect(getByText("Wind direction: S")).toHaveClass(
      "forecast-details__direction"
    );
  });
});
