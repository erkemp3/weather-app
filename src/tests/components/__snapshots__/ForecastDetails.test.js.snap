// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thursday 1st January 1970
    </div>
    <div
      class="forecast-details__temperature"
    >
      Max temp: 22°C
    </div>
    <div
      class="forecast-details__temperature"
    >
      Min temp: 12°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 25%
    </div>
    <div
      class="forecast-details__wind"
    >
      Wind speed: 15mph
    </div>
    <div
      class="forecast-details__direction"
    >
      Wind direction: S
    </div>
  </div>
</DocumentFragment>
`;
